sdks/python/apache_beam/transforms/core.py

```diff
@@ -189,6 +276,54 @@ def finish_bundle(self):
     """
     pass
 
+  def new_tracker(self, restriction):
```

Comment:
It's unclear that all of these methods are only used in the splitable case. Maybe some documentation? (Maybe the names should all reflect this, e.g. containing the word "splittable"?) 

I might go so far as to put these in a mixin class (interface) that would be checked for if the splittable argument was used, which we could also then to enforce that they're implemented.